x-team/auto-ui
# Upgrade ApplicantGridHeader based on new design

**Release Type:** *breaking changes*

Fixes https://x-team-internal.atlassian.net/browse/XP-2455

## Description

The UI components covers all the visual states required to fulfil the **Acceptance Criteria**:

- [x] User must see a drop-down when clicking the filter icon to the right of the column title
- [x] User must still be able to re-order by that column by clicking the column title
- [x] User must be able to filter the results by interacting with drop-down's content inputs
- [x] User must see any columns with active filters have the column title in red
- [x] User must also see a “Clear“ text to be used as a button to clear the filter selection (this is not currently in the design, but is required - best placed at the bottom of the drop down aligned to the left)

## Checklist


## Related PRs

branch | PR
------ | ------
`XP-2460-redux-actions-tabular-filters` | [Auto](https://github.com/x-team/auto/pull/1633)
`XP-2454-filter-timezones` | [Auto-API](https://github.com/x-team/auto-api/pull/553)

## Steps to Test or Reproduce

Although is good to check the impacted component's showcases, this PR will be better tested directly in the application http://localhost:8000/admin/search/lists/none?display=tabular using `auto-api` and `auto` related PRs.

## Impacted Areas in Application

- ApplicantBadge
- ApplicantGrid
- ApplicantGridHeader
- ListsScreen
- SelectBox

## Screenshots

<img width="1679" alt="screen shot 2018-11-08 at 9 44 16 pm" src="https://user-images.githubusercontent.com/131859/48234166-a0178000-e39f-11e8-96cb-9b1461e49092.png">
<img width="1680" alt="screen shot 2018-11-08 at 9 47 46 pm" src="https://user-images.githubusercontent.com/131859/48234261-08fef800-e3a0-11e8-82b1-f526033b7318.png">
<img width="1679" alt="screen shot 2018-11-08 at 9 48 01 pm" src="https://user-images.githubusercontent.com/131859/48234262-09978e80-e3a0-11e8-94d4-7234df9d27ed.png">
<img width="1680" alt="screen shot 2018-11-08 at 9 48 17 pm" src="https://user-images.githubusercontent.com/131859/48234263-09978e80-e3a0-11e8-8daf-e647ea27a414.png">
